### DISCUSSION TITLE
Places API Integration throws tons of Errors

### DISCUSSION THREAD

**Discussion by GeBon22** [neuefische/questions#410] [Web] [answered]
**!
[Link to Code
](https://github.com/GeBon22/Capstone-Project/blob/feature/maps-integration/components/Map.js)!**

Hi there,

when implementing the google places api into my google maps component, I would like to display all lawyers within a 5km range of the users own location. 
But instead I get this Error:
![grafik](https://user-images.githubusercontent.com/114496506/208097413-7da78d54-9f4a-4d21-9b94-c061c1fd9f16.png)

When leaving out the lawyers.map function, I get the following error: 
![grafik](https://user-images.githubusercontent.com/114496506/208097549-40b4ee43-13d6-45ea-ab9d-956f9b7d050c.png)

What I have tried so far: 

- use a different default location (as I am currently quite remote and I wanted to minimize the danger of receiving an error, because there might be no lawyers around here) - did not work - even in Leipzig or Hamburg it does not display the wished results.
- calling the loadLawyers function after fetching the data necessary to display the user location ('center' - state)  - nothing changed

Obviously there is something major here that I am getting wrong. As I worked on this since yesterday noon, I am looking forward to your suggestions and code reviews. 

Cheers

**Comment by sashameniakin** [comment]
Hallo Georg, 

Im not sure, but I would like to try next condition: 
You can try it like this:  
"{lawyers && lawyers.map(..."

Maybe your "lawyers" state is not existing at the call moment...

**Comment by actyralf** [comment, marked as answer]
I think this is the right direction. In your `useEffect` you have the following line: `setLawyers(position.coords.latitude, position.coords.longitude);`
With this directive you are setting your `lawyers` state variable to a numeric value (whatever is in `position.coords.latitude`) and it's no longer an array. When you then try to call `lawyers.map` it will fail because only arrays have the `.map()` method.

**Reply by GeBon22** [reply]
This here is the right direction - as the loadLawyer function was called before the 'center' state was done getting fetched and set, the function and subsequently the mapping would fail throwing different errors. 
This is best avoided using a useEffect or having the map rendered only after the center was set. 
thanks for all answers and help :-)
